common/formats/cue.cpp

```diff
@@ -312,18 +314,62 @@ void CueSheet::parseTracksContext(const char *line) {
 			flags |= lookupInTable(trackFlags, flag.c_str());
 		}
 
-		_files[_currentFile].tracks[_currentTrack].flags = flags;
+		_tracks[_currentTrack].flags = flags;
 
-		debug(5, "Track flags: %d", _files[_currentFile].tracks[_currentTrack].flags);
+		debug(5, "Track flags: %d", _tracks[_currentTrack].flags);
 	} else if (command == "FILE") {
 		parseHeaderContext(line);
 	} else if (command == "PERFORMER") {
-		_files[_currentFile].tracks[_currentTrack].performer = nexttok(s, &s);
+		_tracks[_currentTrack].performer = nexttok(s, &s);
 
-		debug(5, "Track performer: %s", _files[_currentFile].tracks[_currentTrack].performer.c_str());
+		debug(5, "Track performer: %s", _tracks[_currentTrack].performer.c_str());
 	} else {
 		warning("CueSheet: Unprocessed track command %s at line %d", command.c_str(), _lineNum);
 	}
 }
 
+Array<CueSheet::CueFile> CueSheet::files() {
+	return _files;
+}
+
+Array<CueSheet::CueTrack> CueSheet::tracks() {
+	return _tracks;
+}
+
+CueSheet::CueTrack *CueSheet::getTrack(int tracknum) {
+	for (int i = 0; i < _tracks.size(); i++) {
+		if (_tracks[i].number == tracknum) {
+			return &_tracks[i];
+		}
+	}
+
+	return nullptr;
+}
+
+CueSheet::CueTrack *CueSheet::getTrackAtFrame(int frame) {
+	for (int i = 0; i < _tracks.size(); i++) {
+		// Inside pregap
+		if (frame >= _tracks[i].indices[0] && frame < _tracks[i].indices.back()) {
+			debug(5, "CueSheet::getTrackAtFrame: Returning track %i (pregap)", i);
+			return &_tracks[i];
+		}
+
+		// Between index 1 and the start of the subsequent track
+		if (i < _tracks.size() && frame > _tracks[i].indices.back() && frame < _tracks[i+1].indices[0]) {
+			debug(5, "CueSheet::getTrackAtFrame: Returning track %i (inside content)", i);
+			return &_tracks[i];
+		}
+	}
+
+	// Not found within any tracks, but could be in the final track
+	if (frame > _tracks.back().indices.back()) {
+		debug(5, "CueSheet::getTrackAtFrame: Returning final track");
+		return &_tracks.back();
+	}
+
+	// Still not found; could indicate a gap between indices
+	debug(5, "CueSheet::getTrackAtFrame: Not returning a track");
```

Comment:
Same here. Does it make sense to turn it into a warning()? Especially for the cases when we get problems reported by the users.

Reply:
Yes, good point - this one definitely wants to be a warning.